opencti-platform/opencti-front/src/private/components/settings/users/edition/UserEditionConfidence.tsx

```diff
@@ -0,0 +1,213 @@
+    .max(100, t('The value must be less than or equal to 100'))
+    .when('user_confidence_level_enabled', {
+      is: true,
+      then: (schema) => schema.required(t('This field is required')).nullable(),
+      otherwise: (schema) => schema.nullable(),
+    });
+
+  return Yup.object().shape({
+    user_confidence_level_enabled: Yup.boolean(),
+    user_confidence_level: maxConfidenceValidator,
+    overrides: Yup.array().of(
+      Yup.object().shape({
+        entity_type: Yup.string(),
+        max_confidence: maxConfidenceValidator,
+      }),
+    ),
+  });
+};
+
+const UserEditionConfidence: FunctionComponent<UserEditionConfidenceProps> = ({ user, context }) => {
+  const { t_i18n } = useFormatter();
+
+  const initialValues: ConfidenceFormData = {
+    user_confidence_level_enabled: !!user.user_confidence_level,
+    user_confidence_level: user.user_confidence_level?.max_confidence,
+    overrides: user.user_confidence_level?.overrides?.map((override) => ({
+      ...override,
+      max_confidence: override.max_confidence.toString(),
+    })) ?? [] };
+
+  const [commitFieldPatch] = useApiMutation(userMutationFieldPatch);
+
+  const handleSubmitMaxConfidence = (name: string, value: string | null) => {
+    userConfidenceValidation(t_i18n)
+      .validateAt(name, { [name]: value })
+      .then(() => {
+        if (name === 'user_confidence_level') {
+          if (user.user_confidence_level && isNotEmptyField(value)) {
+            // We edit an existing value inside the object: use object_path
+            commitFieldPatch({
+              variables: {
+                id: user.id,
+                input: {
+                  key: 'user_confidence_level',
+                  object_path: '/user_confidence_level/max_confidence',
+                  value: parseInt(value, 10),
+                },
+              },
+            });
+          } else if (!user.user_confidence_level && value) {
+            // We have no user_confidence_level and we add one: push a complete object
+            commitFieldPatch({
+              variables: {
+                id: user.id,
+                input: {
+                  key: 'user_confidence_level',
+                  object_path: '/user_confidence_level/max_confidence',
+                  value: {
+                    max_confidence: parseInt(value, 10),
+                    overrides: [],
+                  },
+                },
+              },
+            });
+          } else if (user.user_confidence_level && !value) {
+            // we have an existing value but we want to remove it: push [null] (and not null!)
+            commitFieldPatch({
+              variables: {
+                id: user.id,
+                input: {
+                  key: 'user_confidence_level',
+                  object_path: '/user_confidence_level/max_confidence',
+                  value: [null],
+                },
+              },
+            });
+          }
```

Comment:
Only the value changes between those cases. Compute them in the ifs and make only one call to `commitFieldPatch` after